### ISSUE TITLE
[C++] Inherited virtual functions are not listed as being inherited in child class docs

### ISSUE THREAD

**Issue by seemann5** [doxygen/doxygen#10081]
## The problem

Publically inherited public functions are normally listed in the **Public Member Functions inherited from parent class** section. This behavior does not work as expected when the parent function is marked as **virtual**, in which case the parent function ends up listed in the main **Public Member Functions** section, without mentioning the inheritance relation, which looks a bit confusing.

## Screenshots

- The non-virtual case that works as expected:
![doxygen_mwe_case_ok](https://github.com/doxygen/doxygen/assets/39963973/4f41399b-c7f7-49e1-a004-0b9188c4ae34)
- The virtual case that does not work as expected:
![doxygen_mwe_case_fail](https://github.com/doxygen/doxygen/assets/39963973/c4139f20-a51a-4366-8641-4d4b4877f89b)

## To Reproduce

Using doxygen built from source, version
```
$ doxygen -v
1.9.8 (04693545bc3b41e42a8c1a2d4a5d0ae036bf23b4)
```
on WSL, running Ubuntu 20.04.6.
The above screenshots are generated as:
```
$ touch a.hpp
$ vim a.hpp (<- here inserting the below file)
$ doxygen -g (fresh doxyfile with default options)
$ doxygen
```
where `a.hpp` consists of
```C++
// The satisfying case

class A_ok {
public:
    /*! \brief the A_ok::f_ok() method is shown in the documentation of B_ok as inherited from A_ok */
    void f_ok();
};
class B_ok : public A_ok {
public:
    /*! \brief the override B_ok::f_ok() */
    void f_ok();
};

// The problematic case

class A_fail {
public:
    /*! \brief the A_fail::f_fail() method is shown in the documentation of B_fail as *not* inherited from A_fail */
    virtual void f_fail();
};
class B_fail : public A_fail {
public:
    /*! \brief the override B_fail::f_fail() */
    void f_fail();
};
```

## Extra information

I tried playing around with marking `B_fail::f_fail()` as either
```C++
void f_fail();
virtual void f_fail();
virtual void f_fail() override;
void f_fail() override;
```
but this doesn't seem to solve the issue.

**Comment by seemann5** [comment]
# Update

Adding a *documented* constructor to the parent class, so the class `A_fail` in the above example, solves the problem.
This works well as a quick fix, but is slightly annoying for pure interfaces.

**Comment by doxygen** [comment]
Interesting finding, seems to be broken like this since 1.9.3 already.

**Comment by doxygen** [comment]
@seemann5 Please verify if the referenced commit fixes the problem for you. Don't close the issue as this will be done automatically when the next version of doxygen is release.

Note that `A_fail`'s `f_fail()` should actually not show up as a method of `B_fail` as it has been reimplemented by `B_fail`. This is now corrected.

**Comment by albert-github** [comment]
@doxygen I think we can have a nullptr problem here:
```
    int count = cd->countMembersIncludingGrouped(m_listType,inheritedFrom,TRUE);
    if ( cd && !optimizeVhdl && count>0 )
```
should probably be:
```
    int count = (cd ? cd->countMembersIncludingGrouped(m_listType,inheritedFrom,TRUE) : 0);
    if ( !optimizeVhdl && count>0 )
```

**Comment by doxygen** [comment]
@albert-github Good catch. I isolated the counter for debugging purposes, reverted that part of the change now.

**Comment by seemann5** [comment]
I just pulled, the new version works as expected! Thanks a lot!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
